test/kdtree/test_kdtree.cpp

```diff
@@ -75,6 +75,7 @@ init ()
     for (float y = -0.5f; y <= 0.5f; y += resolution)
       for (float x = -0.5f; x <= 0.5f; x += resolution)
         cloud.emplace_back(x, y, z);
+  cloud.emplace_back(std::numeric_limits<float>::infinity(), 0, 0);
```

Comment:
Is this needed? Also it doesn't make sense because how can x be inf but not y and z.

Reply:
This point, along with an infinite query point, is what causes the original crash in #4345. A point created from stereo reconstruction with 0 disparity can have a single infinite dimension. Besides, the point isn't meant to be realistic, it's meant to trigger the crash.